crud/select.lua

```diff
@@ -30,6 +30,15 @@ function checkers.vshard_call_mode(p)
 end
 
 local function select_on_storage(space_name, index_id, conditions, opts)
+    -- For compatibility with IPROTO_CALL_RET_TUPLE_EXTENSION feature.
+    if opts ~= nil then
+        if box.tuple.is(opts.scan_value) then
+            opts.scan_value = opts.scan_value:totable()
+        end
+        if box.tuple.is(opts.after_tuple) then
+            opts.after_tuple = opts.after_tuple:totable()
+        end
+    end
```

Comment:
Is it okay that we modify the options table here, or should I make a copy of it?

Reply:
I think we also need to cover external API here:

https://github.com/tarantool/crud/blob/588c215da3699a3c0925fc8b2c0f9aa3640c48c9/crud/select/compat/select_old.lua#L421
https://github.com/tarantool/crud/blob/588c215da3699a3c0925fc8b2c0f9aa3640c48c9/crud/select/compat/select_old.lua#L228

(`compat/select.lua` already covers this, `compat/select_old.lua` not cover it yet.)

Reply:
> I think we also need to cover external API here:

AFAIC, crud's embedded tuple merger can only be used with Tarantool 1.10, which will never have tuples over IPROTO, so we don't need to bother about it.

Reply:
Well, ok

Reply:
> The other question is do we really have to `totable` this one? If the only reason is `checks` (`dev_checks`), then you can simply use `?table|cdata` here.

I also considered this, but I preferred conversion, because we can't check the `cdata` specific type. Though if this function is only internal, I guess it would be fine (but why do we do the type checking then in the first place?) — leaving this is up to you.

I am also not sure that passing the `cdata` down won't require patching a lot of other places downstream. If you insist, I can investigate that.

Reply:
> because we can't check the cdata specific type

It's fine since we don't check table specific type either (we expect an array, but may receive a map).

> I am also not sure that passing the cdata down won't require patching a lot of other places downstream.

Try to patch only `dev_checks` first. If there are any more changes required that are more than two minutes of work, let's stick to your implementation.

Reply:
Looks like it works and we can proceed with your solution.